src/pheromones/hivemind-host/src/RequestDTO.cpp

```diff
@@ -15,26 +21,45 @@ RequestDTO::RequestDTO(const Request& request) : m_id(request.id) {
 RequestDTO::RequestDTO(uint32_t id, const UserCallRequestDTO& request) :
     m_id(id), m_request(request) {}
 
-const std::variant<std::monostate, UserCallRequestDTO>& RequestDTO::getRequest() const {
+RequestDTO::RequestDTO(uint32_t id, const HiveMindApiRequestDTO& request) :
+    m_id(id), m_request(request) {}
+
+RequestDTO::RequestDTO(uint32_t id, const SwarmApiRequestDTO& request) :
+    m_id(id), m_request(request) {}
+
+const std::variant<std::monostate, UserCallRequestDTO, HiveMindApiRequestDTO, SwarmApiRequestDTO>&
+RequestDTO::getRequest() const {
     return m_request;
 }
 
 uint32_t RequestDTO::getId() const { return m_id; }
 
 void RequestDTO::setId(uint32_t id) { m_id = id; }
 
-void RequestDTO::setRequest(const std::variant<std::monostate, UserCallRequestDTO>& request) {
+void RequestDTO::setRequest(const std::variant<std::monostate,
```

Comment:
Could a typedef be usefull for the long
```c
 std::variant<std::monostate, UserCallRequestDTO,HiveMindApiRequestDTO,SwarmApiRequestDTO>
```
type? Similarly for the response DTOs

Reply:
Yes, but at the same time it helps to see if you missed something on the hivemind side. The compiler will throw an error if a type was added on propolis but you are not handling it in the hivemind.